### PR TITLE
Non-modal form for converting travel board request type to virtual

### LINKED ISSUE #15157: Create a non-modal form for converting a legacy appeal travel board hearing request to a virtual hearing request
Connects #14857

## Description
Create a non-modal view/form for converting a legacy appeal travel board hearing request to a virtual hearing request. The form will display the RO queue that the veteran will appear in and the appellant and poa name and email, as seen in [the design](https://www.figma.com/proto/V87TZArfdurCGJiEjQ73ES/Virtual-Hearings?node-id=6333%3A2394&viewport=-839%2C837%2C0.08404029160737991&scaling=min-zoom).

## Out of scope
- Hold off on connecting this view to the change hearing request type action described in #15156
- The textarea for recording notes appears in the design, but is out of scope for this ticket
- The missing email alert on other frames in the Figma link (#15056)

## Acceptance criteria
- [ ] display the scheduling queue that the veteran/case will appear in
- [ ] display the veteran/appellant and poa name and email
- [ ] update Storybook and jest tests

## Design
[Figma link](https://www.figma.com/proto/V87TZArfdurCGJiEjQ73ES/Virtual-Hearings?node-id=6333%3A2394&viewport=-839%2C837%2C0.08404029160737991&scaling=min-zoom)

### PR DESCRIPTION
**Pull request by rubaiyat22**
Resolves #15157 

### Description
- serialize regional office label
- create new component `HearingTypeConversionForm` to display form to convert hearing type to virtual
- write stories for `HearingTypeConversionForm`
- rename method `getAppellantTitleForHearing` to the more general `getAppellantTitle`
- minor changes to  components `AppellantSection` and `VirtualHearingSecion`
- write jest tests for `HearingTypeConversionForm`

### Testing Plan
1. run `make storybook`
2. go to storybook and see `HearingConversionTypeForm` 

### Storybook Story
*For Frontend (Presentationa) Components*
* [X] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [X] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [X] Write a separate story (within the same file) for each discrete variation of the component

### CODE REVIEW

File: client/app/queue/utils.js [hunk 1]
```diff
@@ -330,6 +330,7 @@ export const prepareAppealForStore = (appeals) => {
       veteranGender: appeal.attributes.veteran_gender,
       veteranAddress: appeal.attributes.veteran_address,
       closestRegionalOffice: appeal.attributes.closest_regional_office,
+      closestRegionalOfficeLabel: appeal.attributes.closest_regional_office_label,
```
Comment (rubaiyat22) [inline client/app/queue/utils.js:333]: adding the new serialized field to redux store

File: client/app/hearings/utils.js [hunk 2]
```diff
@@ -110,10 +110,10 @@ export const APPELLANT_TITLE = 'Appellant';
 
 /**
  * Gets the title to use for the appellant of a hearing.
- * @param {object} hearing -- A hearing
+ * @param {string} appellantIsNotVeteran -- bool
  */
-export const getAppellantTitleForHearing = (hearing) =>
-  hearing?.appellantIsNotVeteran ? APPELLANT_TITLE : VETERAN_TITLE;
+export const getAppellantTitle = (appellantIsNotVeteran) =>
```
Comment (rubaiyat22) [inline client/app/hearings/utils.js:115]: renaming method and changing param

File: client/app/hearings/components/Details.jsx [hunk 3]
```diff
@@ -15,7 +15,7 @@ import {
   RESET_VIRTUAL_HEARING
 } from '../contexts/HearingsFormContext';
 import { HearingsUserContext } from '../contexts/HearingsUserContext';
-import { deepDiff, pollVirtualHearingData, getChanges, getAppellantTitleForHearing } from '../utils';
+import { deepDiff, pollVirtualHearingData, getChanges, getAppellantTitle } from '../utils';
```
Comment (rubaiyat22) [inline client/app/hearings/components/Details.jsx:18]: only a name change in this file

File: client/app/hearings/components/ScheduleVeteran.jsx [hunk 4]
```diff
@@ -15,7 +15,7 @@ import { formatDateStr } from '../../util/DateUtil';
 import Alert from '../../components/Alert';
 import { marginTop, regionalOfficeSection, saveButton, cancelButton } from './details/style';
 import { find } from 'lodash';
-import { getAppellantTitleForHearing } from '../utils';
+import { getAppellantTitle } from '../utils';
```
Comment (rubaiyat22) [inline client/app/hearings/components/ScheduleVeteran.jsx:18]: only a name change in this file

File: client/app/hearings/components/VirtualHearingModal.jsx [hunk 5]
```diff
@@ -4,7 +4,7 @@ import React, { useEffect, useState } from 'react';
 import _ from 'lodash';
 import moment from 'moment-timezone';
 
-import { getAppellantTitleForHearing, zoneName } from '../utils';
+import { getAppellantTitle, zoneName } from '../utils';
```
Comment (rubaiyat22) [inline client/app/hearings/components/VirtualHearingModal.jsx:7]: only a name change in this file

File: client/app/hearings/components/VirtualHearings/AppellantSection.jsx [hunk 6]
```diff
@@ -18,50 +19,64 @@ export const AppellantSection = ({
...
+  showOnlyAppellantName
+}) => {
+  const appellantName = hearing?.appellantFullName ? hearing?.appellantFullName :
+    `${hearing?.veteranFirstName} ${hearing?.veteranLastName}`;
+
+  const showTimezoneField = virtual && !video;
+
+  return (
+    <VirtualHearingSection label={appellantTitle} showDivider={showDivider}>
+      {showOnlyAppellantName ? (
```
Comment (rubaiyat22) [inline client/app/hearings/components/VirtualHearings/AppellantSection.jsx:34]: add logic to only display appellant name under `Veteran/Appellant Name` header

File: client/app/hearings/components/VirtualHearings/RepresentativeSection.jsx [hunk 7]
```diff
@@ -23,7 +23,7 @@ export const RepresentativeSection = ({
   appellantTitle
 }) => (
   <VirtualHearingSection label="Power of Attorney">
-    {hearing.representative ? (
+    {hearing?.representative ? (
```
Comment (rubaiyat22) [inline client/app/hearings/components/VirtualHearings/RepresentativeSection.jsx:26]: only adding few safe assigns in this file

File: client/app/hearings/components/VirtualHearings/Section.jsx [hunk 8]
```diff
@@ -1,22 +1,26 @@
...
 import PropTypes from 'prop-types';
 
-export const VirtualHearingSection = ({ label, children, hide }) =>
+import { marginTop } from '../details/style';
+
+export const VirtualHearingSection = ({ label, children, hide, showDivider }) =>
   !hide && (
     <React.Fragment>
-      <div className="cf-help-divider" />
+      {showDivider ? <div className="cf-help-divider" /> : <div {...marginTop(30)} />}
```
Comment (rubaiyat22) [inline client/app/hearings/components/VirtualHearings/Section.jsx:9]: determines whether to show a divider on top

File: client/app/hearings/components/details/VirtualHearingForm.jsx [hunk 9]
```diff
@@ -4,7 +4,7 @@ import React, { useContext, useEffect } from 'react';
 import { ContentSection } from '../../../components/ContentSection';
 import { HearingLinks } from './HearingLinks';
 import { HearingsUserContext } from '../../contexts/HearingsUserContext';
-import { getAppellantTitleForHearing } from '../../utils';
+import { getAppellantTitle } from '../../utils';
```
Comment (rubaiyat22) [inline client/app/hearings/components/details/VirtualHearingForm.jsx:7]: only a name change in this file

File: client/test/app/hearings/components/VirtualHearings/RepresentativeSection.test.js [hunk 10]
```diff
@@ -10,7 +10,7 @@ import { AddressLine } from 'app/hearings/components/details/Address';
 import { VirtualHearingEmail } from 'app/hearings/components/VirtualHearings/Emails';
 import { Timezone } from 'app/hearings/components/VirtualHearings/Timezone';
 import { ReadOnly } from 'app/hearings/components/details/ReadOnly';
-import { getAppellantTitleForHearing } from 'app/hearings/utils';
+import { getAppellantTitle } from 'app/hearings/utils';
```
Comment (rubaiyat22) [inline client/test/app/hearings/components/VirtualHearings/RepresentativeSection.test.js:13]: only a name change on this file

File: client/app/hearings/components/HearingConversion.jsx [hunk 11]
```diff
@@ -12,7 +12,7 @@ import { VirtualHearingSection } from './VirtualHearings/Section';
 import { ReadOnly } from './details/ReadOnly';
 import { HelperText } from './VirtualHearings/HelperText';
 import { HearingTime } from './modalForms/HearingTime';
-import { getAppellantTitleForHearing } from '../utils';
+import { getAppellantTitle } from '../utils';
```
Comment (rubaiyat22) [inline client/app/hearings/components/HearingConversion.jsx:15]: only a name change in this file

File: client/app/hearings/components/HearingTypeConversionForm.jsx [hunk 12]
```diff
@@ -0,0 +1,100 @@
+import React, { useState } from 'react';
```
Comment (rubaiyat22) [inline client/app/hearings/components/HearingTypeConversionForm.jsx:1]: new component

File: client/app/hearings/components/HearingTypeConversionForm.stories.js [hunk 13]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
```
Comment (rubaiyat22) [inline client/app/hearings/components/HearingTypeConversionForm.stories.js:1]: new storybook component

File: client/app/hearings/components/HearingTypeConversionForm.test.js [hunk 14]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
```
Comment (rubaiyat22) [inline client/app/hearings/components/HearingTypeConversionForm.test.js:1]: new jest file

**Review by ferristseng — APPROVED** [review]
LGTM with some minor suggestions!

File: client/app/hearings/components/HearingTypeConversionForm.jsx [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+  return (
+    <React.Fragment>
+      <AppSegment filledBackground>
+        <h1 className="cf-margin-bottom-0">{convertTitle}</h1>
+        <p dangerouslySetInnerHTML={
+          { __html: sprintf(COPY.CONVERT_HEARING_TYPE_SUBTITLE, appeal?.closestRegionalOfficeLabel) }
+        }
+        />
+        <HelperText label={COPY.CONVERT_HEARING_TYPE_SUBTITLE_2} />
+        <AppellantSection {...sectionProps} />
```
[the remark this answers is not in the document]
Comment (rubaiyat22) [reply client/app/hearings/components/HearingTypeConversionForm.jsx:64]: async [discusssion](https://dsva.slack.com/archives/C3EAF3Q15/p1600116202201700?thread_ts=1600092053.191800&cid=C3EAF3Q15)

**Pull request merged**